viewing_party/party.py

```diff
@@ -0,0 +1,201 @@
+            friend_watched_list.append(friend_movie)
+    for user_movie in user_watched_list:
+        if user_movie not in friend_watched_list:
+            unique_user_watch_list.append(user_movie)
+            
+    return unique_user_watch_list
+
+
+# returns a list of movies that the friends have watched, but the user hasn't
+def get_friends_unique_watched(user_data):
+    # compare the user's 'watched' list and the users 'watched' list
+    unique_friends_watch_list= []
+    for friend in user_data['friends']:
+        for movie in friend['watched']:
+            if movie not in user_data['watched']:
+                if movie not in unique_friends_watch_list:
+                    unique_friends_watch_list.append(movie)
+    return unique_friends_watch_list
+    
+# wave 4
+
+def get_available_recs(user_data):
+    rec_list = []
+    friend_watched = []
+    user_watched_titles = []
+    # get to iteration over  subs list and friends hosts
+    for friend in user_data['friends']:
+        for movie in friend['watched']: 
+            friend_watched.append(movie)
+    for movie in user_data['watched']:
+        user_watched_titles.append(movie['title'])
+    for friend_movie in friend_watched:
+        friend_host = friend_movie['host']
+        # if statements that have requirements of the movies to be added to a list    
+        if friend_host in user_data['subscriptions']:
+            if friend_movie['title'] not in user_watched_titles:
+                if friend_movie not in rec_list:
+                    rec_list.append(friend_movie)
+    return rec_list
+
+
+    #WAVE 5
+# get recommendations by genre
+def get_new_rec_by_genre(user_data):
+    #compare most watcehd to friends 
+    most_watched_genre = get_most_watched_genre(user_data)
+    genre_rec_list = []
+    friend_watched = []
+    for friend in user_data['friends']:
+        for movie in friend['watched']:
+            if movie not in friend_watched:
+                friend_watched.append(movie)
+    # add a movie to a list to give recs
+    for friend_movie in friend_watched:
+        friend_genre = friend_movie['genre']
+        if friend_movie not in user_data['watched']:
+            if friend_genre == most_watched_genre:
+                if movie not in genre_rec_list:
+                    genre_rec_list.append(friend_movie)
+    return genre_rec_list
+
+
+# get a list of users faves
+def get_rec_from_favorites(user_data):
+    rec_from_faves_list = []
+    friend_watched = []
+    for friend in user_data['friends']:
+        for friend_movie in friend['watched']:
+            if friend_movie not in friend_watched:
+                friend_watched.append(friend_movie)
+
+    for fave_movie in user_data['favorites']:
+        if fave_movie not in  friend_watched:
+            if fave_movie not in rec_from_faves_list:
+                rec_from_faves_list.append(fave_movie)
+    return rec_from_faves_list
+
+
+
+
+
+
+
+    
+
+    
+
+
+
+
+
+
+
```

Comment:
```suggestion
```

You don't need all of this whitespace. To make your project neater and follow pep8 guidelines, you can have VS Code autoformat your files by right-clicking on a file in VS Code and select Format Document With... then select python from the drop-down menu. You'll then be prompted to install autopep8, click yes and the package will be installed within your virtual environment.